[jasmiinee] iP

jasmiinee opened the pull request:
### Taskon

> "Efficiency is doing better what is already being done." – Peter Drucker [source](https://cioviews.com/top-10-peter-drucker-quotes-for-business-leaders/)

Taskon empowers you to manage your tasks efficiently, so you can focus on what _truly_ matters. It's:

- intuitive
- simple to use
- lightning fast ⚡

All you need to do is:

1. [Download it from here](https://github.com/jasmiinee/ip).
2. Double-click the app.
3. Add your tasks.
4. Let Taskon handle the rest. 😉

And it's **FREE**!

Features:

- [X] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```java
public static void main(String[] args) {
    new Taskon("./data/taskon.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jasmiinee committed:
Rename main class

jasmiinee committed:
Add Greeting

jasmiinee committed:
Add Support for Echo

jasmiinee committed:
Add Support for List

jasmiinee committed:
Add Task class

jasmiinee committed:
Add ability to mark and unmark tasks

jasmiinee committed:
Add Deadline, Event and Todo classes

jasmiinee committed:
Add Support for Deadline, Event and Todo tasks

jasmiinee committed:
Use input/output redirection technique to semi-automate the testing of Taskon

jasmiinee committed:
Exception Handling for creating task

jasmiinee committed:
Exception Handling for creating task

jasmiinee committed:
Add support for deleting tasks

jasmiinee committed:
Fixed bugs and used Java enums

jasmiinee committed:
Update javadoc comments

jasmiinee committed:
Add FileManager class

jasmiinee committed:
Add support for saving tasks

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-Level-7'

jasmiinee committed:
Add support for reading date and time

jasmiinee committed:
Add command to print tasks on a specific date

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-Level-8'

jasmiinee committed:
Add Ui class to deal with interactions with user

jasmiinee committed:
Add storage and parser class and command class

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-A-MoreOOP'

jasmiinee committed:
Package the classes

jasmiinee committed:
Merge remote-tracking branch 'origin/add-gradle-support'

jasmiinee committed:
Add JUnit test cases

jasmiinee committed:
Add javaDoc comments to code

jasmiinee committed:
Comply with coding standard

jasmiinee committed:
Add support to find task by keyword

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-A-JavaDoc'

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

# Conflicts:
#	src/main/java/taskon/parser/Parser.java
#	src/main/java/taskon/storage/Storage.java

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-Level-9'

jasmiinee committed:
Clean up code

jasmiinee committed:
Comply with style check

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-A-CheckStyle'

jasmiinee committed:
Set up JavaFX

jasmiinee committed:
Create GUI for taskon

jasmiinee committed:
Tweak GUI

jasmiinee committed:
Add JavaDoc comments

jasmiinee committed:
Merge remote-tracking branch 'origin/branch-Level-10'

NicholasT1 commented on src/main/java/taskon/DialogBox.java:
```diff
@@ -0,0 +1,84 @@
+    /**
+     * Flips the dialog box such that the ImageView is on the left and text on the right.
+     */
+    private void flip() {
```
Perhaps this name could be more descriptive?

NicholasT1 commented on src/main/java/taskon/MainWindow.java:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    /**
+     * Creates two dialog boxes, one echoing user input and the other containing Duke's reply and then appends them to
```
Just pointing out that it says Duke there since you might have missed it.

NicholasT1 commented on src/main/java/taskon/common/Messages.java:
I like that you created a separate class for outputs. Definitely makes the code easier to read and to add future features.

NicholasT1 commented on src/main/java/taskon/storage/Storage.java:
```diff
@@ -0,0 +1,154 @@
+     * @param task The Task object to convert.
+     * @return A formatted string representing the Task.
+     */
+    private static String taskToFileString(Task task) {
```
Maybe the name could sound more verb-like, eg. convertToFileString

NicholasT1 reviewed (commented):
Overall, I like how your code is nicely separated into appropriate packages and that you try to reduce boiler-plate code as much as possible. I also like the descriptive javadoc comments. Great Job!

appleshill commented on src/main/java/taskon/Main.java:
```diff
@@ -0,0 +1,31 @@
+    private Taskon taskon = new Taskon();
+
+    @Override
+    public void start(Stage stage) {
```
Since start method includes custom behavior (e.g., FXML loading and injecting the Taskon instance) that is not present in the parent method and its [Javadoc](https://docs.oracle.com/javase/8/javafx/api/javafx/application/Application.html#start-javafx.stage.Stage-), adding a Javadoc could help.

appleshill commented on src/main/java/taskon/storage/Storage.java:
```diff
@@ -0,0 +1,154 @@
+     * @throws TaskonException If the line cannot be parsed into a valid Task.
+     */
+    private static Task parseTask(String line) throws TaskonException {
+        Task task = null;
```
Any reason that this is initalised to null? Could it lead to null pointer exceptions (if I'm understanding the logic here correctly)? An alternative could be to just declare task and then handle the unknown task type by throwing a TaskonException

appleshill commented on src/main/java/taskon/storage/Storage.java:
```diff
@@ -0,0 +1,154 @@
+            taskDescription = "T" + SEPARATOR + taskStatus + SEPARATOR + task.getDescription();
+        } else if (task instanceof Deadline) {
+            taskDescription = "D" + SEPARATOR + taskStatus + SEPARATOR + task.getDescription()
+                    + SEPARATOR + ((Deadline) task).getDate().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
```
Line length is beyond the soft limit, you may want to consider adding a line break

appleshill commented on src/main/java/taskon/task/Task.java:
```diff
@@ -0,0 +1,93 @@
+     */
+    public void markAsUndone() throws TaskonException {
+        if (!isDone) {
+            throw new TaskonException("Hmm... it looks like you've already left this task unchecked.\n" + this + "\n");
```
Line length is nearly at the hard limit, you may want to consider adding a line break

appleshill reviewed (commented):
Generally very well done, nearly no coding standards violations. I also like the use of abstraction and additional functionality provided

jasmiinee committed:
Clean up code

jasmiinee committed:
Add assertions

Task parsing and storage lack validation checks, which can lead to errors with unexpected input or file formats.

Validation ensures that tasks and storage data are correctly formatted, reducing runtime errors and improving reliability.

Implement assertions to check the integrity and correctness of task parsing and storage operations.

Assertions provide early feedback during development, catching potential issues before they impact the application.

jasmiinee committed:
Improve code quality

jasmiinee committed:
Merge pull request #1 from jasmiinee/branch-A-Assertions

Add assertions

jasmiinee committed:
Clean up code

jasmiinee committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/taskon/common/Messages.java
#	src/main/java/taskon/parser/Parser.java

jasmiinee committed:
Merge pull request #3 from jasmiinee/branch-A-CodeQuality

Improve code quality

jasmiinee committed:
Set up CI

jasmiinee committed:
Add support to edit tasks without deleting them

Users must delete a task and re-add it in order to change its details.

Add support to modify task details without deleting them. Allow users to edit the description, deadline, and event times directly through the 'edit' command.

Task editing is needed to streamline user interaction and avoid unnecessary task deletion when making small changes, such as updating a deadline or correcting a description.

jasmiinee committed:
Merge branch 'branch-BCD-Extension'

jasmiinee committed:
Improve the GUI

jasmiinee committed:
Add unique personality

jasmiinee committed:
Improve code to handle event error

jasmiinee committed:
Add more JUnit test

jasmiinee committed:
Clean up code

jasmiinee committed:
Update user guide

jasmiinee committed:
Fix image issues

jasmiinee committed:
Fix image issues

jasmiinee committed:
Update README.md
